app/services/tagging/common_ancestor_finder.rb

```diff
@@ -11,12 +11,16 @@ def find_all
       all_results = filtered_content_enum.map do |content_hash|
         content_id = content_hash['content_id']
         title = content_hash['title']
-        expanded_links_hash = Services.publishing_api.get_expanded_links(content_id).to_h
-        {
-          content_id: content_id,
-          title: title,
-          common_ancestors: find_common_ancestors(taxon_paths(expanded_links_hash))
-        }
+        begin
+          expanded_links_hash = Services.publishing_api.get_expanded_links(content_id).to_h
+          {
+            content_id: content_id,
+            title: title,
+            common_ancestors: find_common_ancestors(taxon_paths(expanded_links_hash))
+          }
+        rescue GdsApi::HTTPNotFound
+          puts("Cannot find content with id: #{content_id}")
+        end
```

Comment:
What will this block evaluate to in the case of an exception? I'm not sure the select on line 21 will work without the result being a dictionary.

Reply:
oops...